src/redux/banner/types.ts

```diff
@@ -15,11 +15,13 @@ export interface BannerActions extends Action<BannerActionType> {
 }
 
 export interface SetBannerAction extends BannerActions {
+  type: BannerActionType.SET_BANNER
   state: BannerState;
 }
 
 export interface BannerState {
-  show: boolean
-  text: string
-  timestamp: string
+  banner: undefined | {
+    text: string | undefined
+    etag: string | null
```

Comment:
I don't like the fact that the BannerState object contains a single property "banner" which again contains the wanted values.

Reply:
I'm also not very happy with that but i could think of another name. Top level redux states must not be undefined.

Reply:
Then just set text to "" and date to new Date(0) and don't remove the show variable?

Reply:
Not sure yet. I need to differentiate between "not fetched yet" and "fetched but ignored". But yeah. Maybe i should just use an empty string. I mean: who would show a banner with no content?

Reply:
We explicitly don't show the banner if the text is empty. At least we did so until now…